src/main/java/com/okta/tools/OktaAwsCliAssumeRole.java

```diff
@@ -94,8 +97,23 @@ private static Path getSessionFilePath() {
 
     String run(Instant startInstant) throws Exception {
         Optional<Session> session = getCurrentSession();
-        if (session.isPresent() && sessionIsActive(startInstant, session.get()))
-            return session.get().profileName;
+        Instant expiry = startInstant.plus(864000 - 30, ChronoUnit.SECONDS);
+        String profileStore = System.getProperty("user.home") + "/.aws/.okta-profile-expiry";
+        Ini ini = new Ini(new File(profileStore));
+
+        if (session.isPresent() && sessionIsActive(startInstant, session.get())) {
+            if (OktaAwsConfig.createAwscli().oktaProfile != null) {
```

Comment:
The use of **OktaAwsConfig.createAwscli().oktaProfile** here is confusing and redundant. You already have the **oktaProfile** field here whose value was established by that exact code being called in the main method.